samples/bluetooth/hci_vs_scan_req/src/main.c

```diff
@@ -0,0 +1,326 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2024 Giancarlo Stasi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/bluetooth/addr.h>
+
+#if !defined(CONFIG_BT_EXT_ADV) && !defined(CONFIG_BT_CTLR_VS_SCAN_REQ_RX)
+#error "Either BT_EXT_ADV or BT_CTLR_VS_SCAN_REQ_RX must be selected to have scan request callback"
```

Comment:
The run test with `qemu_cortex_m3` is triggering this `#error `directive. This seems due to the fact that some configs, including `CONFIG_BT_CTLR_VS_SCAN_REQ_RX`, have been moved in `overlay-bt_ll_sw_split.conf`.
@jori-nordic, do you have any idea about how to make qemu aware of it? Or whatever other fix? (Better not removing this check).

Reply:
> In this particular case, the CI is right. The qemu target compiles with an "off-chip" controller. E.g. over UART. In that case, all the `BT_CTLR` kconfigs are not present. So you can't really assert on them.
> 
> To fix that, you can:
> 
>     * add an early-exit in `main()` if `BT_EXT_ADV` is not present
> 
>     * add error handling + log msg [here](https://github.com/zephyrproject-rtos/zephyr/blob/cf40a7f67519fdd3d2576c33261d08962a947829/samples/bluetooth/hci_vs_scan_req/src/main.c#L136) instructing the user what config to enable on the controller board.
> 
> 
> edit: On re-reading the sample, I would prefer if it was only demonstrating the VS command and not the extended advertising part. It would simplify the code much more and be more focused.

At the end, it was easier to remove these checks. I have similar ones in our application' module managing the advertisements, not really needed in sample app.